src/main/kotlin/io/openfuture/state/controller/WalletController.kt

```diff
@@ -0,0 +1,48 @@
+package io.openfuture.state.controller
+
+import io.openfuture.state.domain.Wallet
+import io.openfuture.state.service.WalletService
+import org.springframework.web.bind.annotation.*
+import java.time.LocalDateTime
+import javax.validation.Valid
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotEmpty
+import javax.validation.constraints.NotNull
+
+@RestController
+@RequestMapping("/api/wallets")
+class WalletController(private val walletService: WalletService) {
+
+    @PostMapping
+    suspend fun save(@RequestBody @Valid request: SaveWalletRequest): WalletDto {
```

Comment:
now it looks much better, but let's define as a styleguide to put Valid prior to RequestBody